client/aggregator.go

```diff
@@ -57,12 +57,10 @@ func (c *watchAggregator) distribute(in <-chan Result, cancel context.CancelFunc
 		c.subscriberLock.Unlock()
 
 		var m Result
+		var ok bool
+
 		select {
-		case res, ok := <-in:
-			if !ok {
-				return
-			}
-			m = res
+		case m, ok = <-in:
```

Comment:
once `!ok`, we'll end up in an infinite loop of `!ok` returns here, no? shouldn't we find a way to exit once the channel is closed?

Reply:
👍 good catch - I guess we can return after `c.subscriberLock.Unlock()` below if `!ok`